composeApp/src/commonMain/kotlin/org/ooni/probe/domain/DeleteTestDescriptor.kt

```diff
@@ -0,0 +1,17 @@
+package org.ooni.probe.domain
+
+import org.ooni.probe.data.models.InstalledTestDescriptorModel
+import org.ooni.probe.data.models.toDescriptor
+import org.ooni.probe.data.repositories.PreferenceRepository
+
+class DeleteTestDescriptor(
+    private val preferencesRepository: PreferenceRepository,
+    private val deleteByRunId: suspend (InstalledTestDescriptorModel.Id) -> Unit,
+) {
+    suspend operator fun invoke(testDescriptor: InstalledTestDescriptorModel) {
+        preferencesRepository.removeDescriptorPreferences(
+            descriptor = testDescriptor.toDescriptor(),
+        )
+        deleteByRunId(testDescriptor.id)
```

Comment:
Shouldn't we be deleting the results/measurements and files related with this descriptor too? 

This way, if the user installs the same link again, they will see the results back. But they will be clogging some space in the meantime. In alternative we can also soft delete the descriptor (hide from the homepage and auto-run, but keep in the DB).

Reply:
We should be deleting associated measurements. I saw you doing some measurements stuff and wanted to call some ready made API's if possible to do so instead of reinventing the wheel